File: server/sockets/packets/handshake.py
from server.sockets.packets import PacketType
from server.sockets.packets.packet import Packet


class HandshakePacket(Packet):
    def __init__(self, data):
        self.name = data['name']

    def dictify(self):
        pass

    def get_type(self):
        return PacketType.Handshake

File: client/src/ts/packets/handshake.ts
import Packet, {PacketType} from './packet';

export default class HandshakePacket extends Packet {
    constructor(private name: string) {
        super(PacketType.Handshake);
    }

    public dictify(): any {
        return {'type': this.packetType, 'name': this.name};
    }
}
